File: 0191-number-of-1-bits/0191-number-of-1-bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        # return str(bin(n)).count('1')
        # converting the number to binary and then to string and count digits with one

# when number is greater than 0, convert to binary i.e divide by 2
        res = 0
        while n > 0:
            # if remainder is not zero then increment res
            if n%2 != 0:
                res += 1
            # get the remaining number
            n //= 2
        return res

File: 0191-number-of-1-bits/0191-number-of-1-bits.java
class Solution {
    public int hammingWeight(int n) {
        int res = 0;
        // while n is not zero 
        while(n!=0){
            // if remainder is not zero then increment res
            if (n%2 != 0)
            res += 1;
            // divide the number n by 2
            n = n >> 1; 
        }
        return res;
    }
}